src/pip/_internal/utils/logging.py

```diff
@@ -106,6 +106,12 @@ def format(self, record):
         by our current indentation level.
         """
         formatted = super(IndentingFormatter, self).format(record)
+        if record.levelno >= logging.WARNING:
```

Comment:
This can probably simplified to be:

```py
if record.levelno >= logging.ERROR:
    ...
elif record.levelno >= logging.WARNING:
    ...
```

Reply:
The reason I did it that way is that if the level is less than `logging.WARNING`, it shouldn't have to be checked a second time (i.e. only if a message is a warning do you have to distinguish between warning and error).

Reply:
Fair enough. FWIW it is likely a very low overhead situation and I'm tempted to holler premature optimization.

---

I personally prefer to keep the indentation levels low, especially checks on the value of the same things, so I'd have done something like:

```py
if record.levelno < logging.WARNING:
    # lower logging levels don't need highlighting
    pass
elif record.levelno >= logging.ERROR:
    # record is either critical or an error
    ...
else:
    # record is a warning
    ...
```

It's not a major thing anyway so use whichever form you prefer. :)

Reply:
```suggestion
        if record.levelno >= logging.WARNING:
            formatted = "{}: {}".format(record.levelname, formatted)
```
should work I think.

Reply:
Nice @xavfernandez!

Reply:
@xavfernandez I had thought of that, too. The reason I didn't do it that way is because I didn't want to have the log messages say `CRITICAL`. I thought `ERROR` was simpler and had a clearer meaning. But if you both like `CRITICAL` we can do that instead.

Thanks for the feedback both of you. I'll make some changes and resubmit. (I noticed something else that should be addressed, too.)

Reply:
PS - @pradyunsg thanks. The `pass` idea is a good one as it achieves both goals of less indentation and not re-checking the common case.